### PR TITLE
Allow js requests from stripe

### PR DESCRIPTION
**Pull request by ifosli**
# 📲 What

Allow js requests to render in our survey webview. This is needed in order for users to be able to add new addresses.

# 👀 See

Trello, screenshots, external resources?

| Before 🐛 | After 🦋 | Entire flow |
| --- | --- | --- |
| Simulator Screen Recording - iPhone 15 Pro Max - 2024-06-25 at 12 34 21 | Simulator Screen Recording - iPhone 15 Pro Max - 2024-06-25 at 12 30 28 | Simulator Screen Recording - iPhone 15 Pro Max - 2024-06-25 at 12 36 13|

# ✅ Acceptance criteria

- [x] New addresses can be added in the survey web view

### CODE REVIEW

File: Library/ViewModels/SurveyResponseViewModel.swift [hunk 1]
```diff
@@ -166,3 +170,7 @@ private func isSurvey(request: URLRequest) -> Bool {
   guard case (.project(_, .survey, _))? = Navigation.match(request) else { return false }
   return true
 }
+
+private func isStripeElement(_ request: URLRequest) -> Bool {
+  return request.url?.host == "js.stripe.com"
```
Comment (amy-at-kickstarter) [inline Library/ViewModels/SurveyResponseViewModel.swift:175]: Can/should we qualify this any more specifically, or is just the host sufficient? I'm not a security engineer but is there any risk to blanket allowing all requests to Stripe's domain?

**Review by amy-at-kickstarter — APPROVED** [review]
LGTM.

File: Library/ViewModels/SurveyResponseViewModel.swift (hunk 1 above)
Comment (ifosli) [reply Library/ViewModels/SurveyResponseViewModel.swift:175]: Good question! Confirmed with web ppl that this is fine: https://kickstarter.slack.com/archives/C06EKGN4MHU/p1719343241132189?thread_ts=1719337090.551589&cid=C06EKGN4MHU

Comment (amy-at-kickstarter) [reply Library/ViewModels/SurveyResponseViewModel.swift:175]: Awesome, thank you for checking!

**Pull request merged**